core/tools/codegen/gen_font.py

```diff
@@ -288,25 +286,36 @@ def _h_file_name(self) -> Path:
 
     def write_files(self) -> None:
         self.write_c_files()
-        self.write_foreign_json()
+        if self.gen_normal:
+            self.write_foreign_json(upper_cased=False)
+        if self.gen_upper:
+            self.write_foreign_json(upper_cased=True)
```

Comment:
Nit: is there any advantage in generating the character twice?

Reply:
I don't see an easy way how to map the same foreign glyphs from one json without changing the mechanics too much. So now if both regular and `_upper` is used, the foreign glyps are duplicated.